include/interface/drivers/RFIDReader.h

```diff
@@ -0,0 +1,119 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <span>
+
+#include "platform/Callback.h"
+
+using tagAvailableCallback = mbed::Callback<void()>;
+
+#include "BufferedSerial.h"
+
+namespace leka {
+
+namespace rfid {
+
+	struct Tag {
+		std::array<uint8_t, 7> id {};
+		std::array<uint8_t, 4> sak {};
+		std::array<uint8_t, 16> data {};
+	};
+
+	enum class Flag : uint8_t
+	{
+		sb_7 = 0b00000111,	 // significant bits : 7
+		sb_8 = 0b00001000,	 // significant bits : 8
+		crc	 = 0b00100000
+	};
+
+	constexpr auto operator|(Flag lhs, Flag rhs) -> Flag
+	{
+		return static_cast<Flag>(static_cast<std::underlying_type_t<Flag>>(lhs) |
+								 static_cast<std::underlying_type_t<Flag>>(rhs));
+	};
+
+	enum class Protocol : uint8_t
+	{
+		ISO14443A,
+		ISO14443B,
+		ISO15693,
+		ISO18092
```

Comment:
```suggestion
		ISO18092,
```

c'est idiot mais si on en rajoute un après, ça évite d'avoir deux lignes qui changent.

dans ce cas on en rajoutera surement pas mais c'est un truc à faire par défaut